src/org/zaproxy/zap/extension/globalexcludeurl/DialogAddToken.java

```diff
@@ -117,6 +123,18 @@ protected void init() {
         getEnabledCheckBox().setSelected(true);
         getDescTextField().setText("");
         token = null;
+        
+		// Handle escape key to close the dialog
+		KeyStroke escape = KeyStroke.getKeyStroke(KeyEvent.VK_ESCAPE, 0, false);
```

Comment:
This should be added in `getFieldsPane` (or the constructors), this method (`init`) is called each time the dialogue is shown. Unless it was just to apply to add dialogue? (In that case it should be added a flag to just do this once.)